packages/palette/src/utils/useSentinelVisibility.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState, useRef } from "react"
+import { Box } from "../elements/Box"
+
+/**
+ * Used to track if an element is scrolled vertically or not
+ */
+export const useSentinelVisibility = () => {
+  const [isSentinelVisible, setSentinelVisibility] = useState(false)
+  const ref = useRef<HTMLElement | null>(null)
+
+  useEffect(() => {
+    if (!("IntersectionObserver" in window)) return // Unsupported
+
+    const element = ref.current
+    if (!element) return
+
+    const observer = new IntersectionObserver(
+      ([entry]) => {
+        setSentinelVisibility(!entry.isIntersecting)
+      },
+      {
+        root: null,
+        rootMargin: "0px",
+        threshold: 1.0,
+      }
+    )
+
+    observer.observe(element)
+
+    return () => {
+      observer.unobserve(element)
+    }
+  }, [isSentinelVisible])
+
+  return {
+    isSentinelVisible,
+    sentinel: (
+      <Box ref={ref as any} position="relative" width="100%" height={0} />
+    ),
+  }
+}
```

Comment:
`useOnScroll` sounds like a hook for attaching something to a scroll handler. Renamed this to be pretty explicit about what it's doing and instead of returning the ref, just return out the whole sentinel.

Reply:
make a lot of sense, thanks for the improvement!

Reply:
I think this has a docs entry